### ISSUE TITLE
Ansi-tests: Methods on gensyms fail

### ISSUE THREAD

**Issue by kpoeck** [drmeister/clasp#429]
````
COMMON-LISP-USER> (LET ((SYM (GENSYM)))
        (VALUES
         (EVAL
           `(DEFMETHOD ,SYM
                       (X)
                       (LIST X)))
         
         (FUNCALL SYM 1)))

Condition of type: PACKAGE-ERROR
Package error on package "NIL"

Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.
````

**Comment by kpoeck** [comment]
error in jit-function-name. Have a fix, will create PR

**Comment by kpoeck** [comment]
Second case:
````
(LET ((SYM (GENSYM)))
        (VALUES
         (EVAL
           `(DEFMETHOD (setf ,SYM)
                       (X)
                       (LIST X)))
         
         (FUNCALL SYM 1)))
````

**Comment by kpoeck** [comment]
Fixed by #430

**Comment by kpoeck** [comment]
Fixed in dev with the merged PR
